public/js/actions/sources.js

```diff
@@ -95,29 +86,30 @@ function newSources(sources) {
  * @static
  */
 function loadSourceMap(generatedSource) {
-  return ({ dispatch, getState }) => {
-    let sourceMap = getSourceMap(getState(), generatedSource.id);
-    if (sourceMap) {
+  return async function({ dispatch, getState }) {
```

Comment:
Yeah, the promise middleware only needs to be used for UI interactions, because the main point of it is optimistic updates. A `start` action is fired when the promise starts which allows the UI to immediately update. The `[PROMISE]` syntax itself is a way of namespacing values for the middleware and is pretty common in redux (at least it used to be).